ServerCore/PuzzleStateHelper.cs

```diff
@@ -475,6 +475,7 @@ public static async Task UpdateTeamsWhoSentResponse(PuzzleServerContext context,
                 var submissionsThatMatchResponse = await (from PuzzleStatePerTeam pspt in context.PuzzleStatePerTeam
                                                           join Submission sub in context.Submissions on pspt.Team equals sub.Team
                                                           where pspt.PuzzleID == response.PuzzleID &&
+                                                          sub.PuzzleID == response.PuzzleID &&
```

Comment:
This is the fix for Mariia's bug - forgot to unstage it. I'll remove if people are concerned.

Reply:
This is fine